src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -513,4 +525,33 @@ private Result shouldSaveResult(@NotNull Result result, boolean shouldSave) {
             return result;
         }
     }
+
+    /**
+     * Retrieves aggregated feedback details for a given exercise, calculating relative counts based on the total number of distinct results.
+     * The task numbers are assigned based on the associated test case names, using the set of tasks fetched from the database.
+     * <br>
+     * For each feedback detail:
+     * 1. The relative count is calculated as a percentage of the total number of distinct results for the exercise.
+     * 2. The task number is determined by matching the test case name with the tasks.
+     *
+     * @param exerciseId The ID of the exercise for which feedback details should be retrieved.
+     * @return A list of FeedbackDetailDTO objects, each containing:
+     *         - feedback count,
+     *         - relative count (as a percentage of distinct results),
+     *         - detail text,
+     *         - test case name,
+     *         - determined task number (based on the test case name).
+     */
+    public List<FeedbackDetailDTO> findAggregatedFeedbackByExerciseId(long exerciseId) {
+        long distinctResultCount = studentParticipationRepository.countDistinctResultsByExerciseId(exerciseId);
```

Comment:
Is this a different value than `feedbackDetails.length`?

If yes please add a comment why we cant use the length, if not, please remove the query

Reply:
Yes its different as feedbackDetails.length represents the rows in our table at the end. For example 1 result can have 2 different distinct feedback then a second result. This would leave to feedbackDetails.length beeing 4 even tho we have 2 results.